src/main/java/com/fasterxml/jackson/core/json/async/NonBlockingJsonParser.java

```diff
@@ -1272,7 +1290,7 @@ protected JsonToken _finishNonStdTokenWithEOF(int type, int matched) throws IOEx
     protected JsonToken _finishErrorToken() throws IOException
     {
         while (_inputPtr < _inputEnd) {
-            int i = (int) _inputBuffer[_inputPtr++];
+            int i = (int) getNextByteFromBuffer();
```

Comment:
... and if that `& 0xFF` is included, we'd likely require alternative of `getNextSignedByteFromBuffer()` or such, to be used here.

Reply:
done